nilearn/regions/tests/test_region_extractor.py

```diff
@@ -149,3 +171,89 @@ def test_region_extractor_fit_and_transform():
+                      [0., 1., 1.],
+                      [0., 0., 0.]],
+                     [[0., 0., 0.],
+                      [1., 0., 0.],
+                      [0., 1., 0.]],
+                     [[0., 0., 1.],
+                      [1., 0., 0.],
+                      [0., 1., 1.]]])
+    sum_data = np.sum(data)
+    # To remove small regions, data should be labelled
+    label_map, n_labels = ndimage.label(data)
+    sum_label_data = np.sum(label_map)
+
+    min_size = 10
+    # data can be act as mask_data to identify regions in label_map because
+    # features in label_map are built upon non-zeros in data
+    index = np.arange(n_labels + 1)
+    removed_data = _remove_small_regions(label_map, data, index, min_size)
+    sum_removed_data = np.sum(removed_data)
+
+    assert_true(sum_removed_data < sum_label_data)
+
+
+def test_connected_label_regions():
+    shape = (13, 11, 12)
+    affine = np.eye(4)
+    n_regions = 9
+    labels_img = testing.generate_labeled_regions(shape, affine=affine,
+                                                  n_regions=n_regions)
+    labels_data = labels_img.get_data()
+    n_labels_wo_reg_ext = len(np.unique(labels_data))
+
+    # region extraction without specifying min_size
+    extracted_regions_on_labels_img = connected_label_regions(labels_img)
+    extracted_regions_labels_data = extracted_regions_on_labels_img.get_data()
+    n_labels_wo_min = len(np.unique(extracted_regions_labels_data))
+
+    assert_true(n_labels_wo_reg_ext < n_labels_wo_min)
+
+    # with specifying min_size
+    extracted_regions_with_min = connected_label_regions(labels_img,
+                                                         min_size=100)
+    extracted_regions_with_min_data = extracted_regions_with_min.get_data()
+    n_labels_with_min = len(np.unique(extracted_regions_with_min_data))
+
+    assert_true(n_labels_wo_min > n_labels_with_min)
+
+    # Test the names of the brain regions given in labels.
+    # Test labels for 9 regions in n_regions
+    labels = ['region_a', 'region_b', 'region_c', 'region_d', 'region_e',
+              'region_f', 'region_g', 'region_h', 'region_i']
+
+    # If labels are provided, first return will contain extracted labels image
+    # and second return will contain list of new names generated based on same
+    # name with assigned on both hemispheres for example.
+    extracted_reg, new_labels = connected_label_regions(labels_img,
+                                                        min_size=100,
+                                                        labels=labels)
+    # new_labels returned must be equal or more than what was in labels.
+    assert_true(new_labels >= len(labels))
+
+    # labels given in numpy array
+    labels = np.asarray(labels)
+    extracted_reg2, new_labels2 = connected_label_regions(labels_img,
+                                                          labels=labels)
+    assert_true(new_labels2 >= len(labels))
```

Comment:
When "labels" is wrong ? You mean when "labels" given are less than number of unique labels in labels_img ?